packages/quantic/force-app/main/default/lwc/quanticTabBar/quanticTabBar.js

```diff
@@ -0,0 +1,354 @@
+    if (this.hasRendered) {
+      if (this.maxMoreButtonWidth < this.moreButtonWidth) {
+        this.maxMoreButtonWidth = this.moreButtonWidth;
+      }
+
+      this.expandedMoreButton =
+        this.containerWidth > this.maxMoreButtonWidth + this.selectedTabWidth;
+    }
+  }
+
+  /**
+   * Updates the visibility of the more button.
+   * @param {boolean} show
+   */
+  updateMoreButtonVisibility(show) {
+    this.moreButton?.style.setProperty('display', show ? 'block' : 'none');
+  }
+
+  /**
+   * Updates the tabs visibility.
+   * @param {Array<Element>} tabElements
+   * @param {boolean} isVisible
+   */
+  updateTabVisibility(tabElements, isVisible) {
+    tabElements.forEach((tab, index) => {
+      const tabsCount = this.getTabsFromSlot().length;
+      // @ts-ignore
+      tab.style.setProperty(
+        'order',
+        isVisible ? index + 1 : tabsCount - tabElements.length + index + 1
+      );
+      // @ts-ignore
+      tab.style.setProperty('visibility', isVisible ? 'visible' : 'hidden');
+    });
+  }
+
+  /**
+   * Indicates whether the tabs are causing an overflow.
+   * @returns {boolean}
+   */
+  get isOverflow() {
+    return this.slotContentWidth >= this.containerWidth;
+  }
+
+  /**
+   * Returns the tab bar container element.
+   * @returns {Element}
+   */
+  get container() {
+    return this.template.querySelector('.tab-bar_container');
+  }
+
+  /**
+   * Returns the container's width.
+   * @returns {number}
+   */
+  get containerWidth() {
+    return this.getAbsoluteWidth(this.container);
+  }
+
+  /**
+   * returns the width of the content of the slot.
+   * @returns {number}
+   */
+  get slotContentWidth() {
+    return this.getTabsFromSlot().reduce(
+      (total, el) => total + this.getAbsoluteWidth(el),
+      0
+    );
+  }
+
+  /**
+   * returns the width of the more button.
+   * @returns {number}
+   */
+  get moreButtonWidth() {
+    return this.moreButton ? this.getAbsoluteWidth(this.moreButton) : 0;
+  }
+
+  /**
+   * returns the width of the currently selected tab.
+   * @returns {number}
+   */
+  get selectedTabWidth() {
+    return this.getAbsoluteWidth(this.selectedTab);
+  }
+
+  /**
+   * Returns the overflowing tabs.
+   * @returns {Array<Element>}
+   */
+  get overflowingTabs() {
+    const containerRelativeRightPosition =
+      this.container.getBoundingClientRect().right;
+    const selectedTabRelativeRightPosition =
+      this.selectedTab.getBoundingClientRect().right;
+
+    return this.getTabsFromSlot().filter((element) => {
+      const tabPositionedBeforeSelectedTab =
+        selectedTabRelativeRightPosition >
+        element.getBoundingClientRect().right;
+      const rightPositionLimit = !this.isOverflow
+        ? containerRelativeRightPosition
+        : tabPositionedBeforeSelectedTab
+        ? containerRelativeRightPosition -
+          this.moreButtonWidth -
+          this.selectedTabWidth
+        : containerRelativeRightPosition - this.moreButtonWidth;
```

Comment:
Some of this should be split into intermediate variables for clarity.

Reply:
updated 👌